src/expiring_hash_map/tests.rs

```diff
@@ -0,0 +1,89 @@
+use super::*;
+use std::task::Poll;
+use tokio02 as tokio;
+use tokio_test::{assert_pending, assert_ready, task};
+
+fn unwrap_ready<T>(poll: Poll<T>) -> T {
```

Comment:
I feel like using https://docs.rs/tokio-test/0.2.0/tokio_test/macro.assert_ready_eq.html would work well here, happy to add more asserts to tokio-test.

Reply:
Ready eq requires specifying a type for expected `Ok`. I don't know how to work around that, but I'm happy to correct it if you show me the way.

Reply:
It shouldn't be working with result types? You can peek the internal impl here https://docs.rs/tokio-test/0.2.0/src/tokio_test/macros.rs.html#174 it just does basically what this does.

Reply:
I do `assert_eq!(unwrap_ready(fut.poll()).unwrap().unwrap().0, "val");`
I can't find a way to do the same with `assert_ready_eq`.
For instance:
- `assert_ready_eq!(fut.unwrap().unwrap().0, "val");` - down't work, cause the first argument has to be the future;
- `assert_ready_ok!(fut.unwrap().0, "val");` - down't work either for the same reason;
- `assert_ready_ok!(fut, Some(("val", ?));` - I'd love to write it in this form, but I don't know the value of the `Key` here, so it doesn't work either.

I'd be happy to improve anything there. The solution I settled on works, but I'm not very pleased with it.